src/main/java/taxi/dao/jdbc/CarJdbcDao.java

```diff
@@ -0,0 +1,199 @@
+                PreparedStatement preparedStatement = connection.prepareStatement(updateQuery)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't update car " + car.getId(), e);
+        }
+        deleteDriversFromCars(car);
+        addDriversToCar(car);
+        return car;
+        // remove old relations between this car(car id) and drivers(from driver_cars table)
+        // delete from cars drivers where car id = carid (from update) -> insert for each
+        // insert new relations using car.getDrivers (71 line) for each driver adding new cars
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE cars SET deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(deleteQuery)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't delete car - id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriverId(Long driverId) {
+        String getAllCarsByDriver = "Select * FROM driver_cars dc"
+                + " INNER JOIN cars c ON dc.driver_id = c.id"
+                + " WHERE dc.driver_id = ? AND c.deleted = false ";
+        List<Car> carsByDriver = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(getAllCarsByDriver)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                carsByDriver.add(getCarFromResult(resultSet));
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant create list of all cars for this driver "
+                    + driverId, e);
+        }
+        for (Car car : carsByDriver) {
+            car.setDrivers(getDriversFromCar(car.getId()));
+        }
+        return carsByDriver;
+    }
+
+    private Car getCarFromResult(ResultSet resultSet) {
+        try {
+            Long carId = resultSet.getObject("id", Long.class);
+            String model = resultSet.getObject("model", String.class);
+            Long manufacturerID = resultSet.getObject("manufacturer_id", Long.class);
+            String manufacturerName = resultSet.getObject("manufacturer_name", String.class);
+            String manufacturerCountry = resultSet.getObject("manufacturer_country", String.class);
```

Comment:
your column names are different in your init_db file

Reply:
Should the column names be changed in this case? otherwise I don't get, how the code will get 2 different id's from resultSet